src/inheritAttributes.js

```diff
@@ -0,0 +1,53 @@
+import { shallowMerge, flatten, getAttributes, parseDuration } from './fns';
+
+export const inheritAttributes = mpd => {
+  const BaseUrl = mpd.getElementsByTagName('BaseURL');
+  const baseUrl = BaseUrl && BaseUrl.length ? BaseUrl[0].innerHTML : '';
+
+  const mpdAttributes = getAttributes(mpd);
+
+  mpdAttributes.sourceDuration = mpdAttributes.mediaPresentationDuration ?
+    parseDuration(mpdAttributes.mediaPresentationDuration) : 0;
+
+  const periods = Array.from(mpd.getElementsByTagName('Period'));
+
+  if (periods.length &&
+      periods.length !== 1) {
+    // TODO add support for multiperiod
+    throw new Error('INVALID_NUMBER_OF_PERIOD');
+  }
+
+  const representationsByPeriod = periods.map((period, periodIndex) => {
```

Comment:
If you wanted to you could probably set up each of these functions for unit testing as well if you extract them/export them from here.  This would let you write specific tests for the anonymous function for representationsByPeriod, as well as the one that is setting representationsByAdaptationSet.  

That might help to make unit testing on inheritAttributes trivia. since you can cover testing a bulk of the logic in the above mentioned functions.